openshift_tools/reporting/zabbixplot.py

```diff
@@ -48,45 +64,52 @@ def get_event_data(self):
         """Retrieve event data from Zabbix API"""
         # Using a double negative because empty arrays are truthy
         if self.events is not False:
-            return self.events
+            return
         # Warning: If there are variables in the description, they won't be filled in
         triggers = self.zapi.trigger.get(group=self.group, search={"description": self.trigger}, selectHosts="extend")
         triggerids = [t["triggerid"] for t in triggers]
-        self.events = self.zapi.event.get(time_from=self.timestamp_start, time_till=self.timestamp_end,
-                                         triggerids=triggerids, limit=self.limit, selectHosts="extend",
-                                         sortfield="clock", sortorder="DESC")
-
-    def __init__(self, server, username, password, group, trigger, days, limit):
-        self.server = server
-        self.username = username
-        self.password = password
+        self.events = self.zapi.event.get(time_from=self.timespan.timestamp_start,
+                                          time_till=self.timespan.timestamp_end, objectids=triggerids,
+                                          limit=self.limit, selectHosts="extend", sortfield="clock", sortorder="DESC")
+        if len(self.events) >= self.limit:
+            print("Warning! Data has been truncated by the result limit. The graph will be inaccurate.",
+                  file=sys.stderr)
+
+    # Reason: All of these parameters are needed
+    # Status: permanently disabled
+    # pylint: disable=too-many-arguments
+    def __init__(self, zapi, group, trigger, days, limit):
+        self.zapi = zapi
         self.group = group
         self.trigger = trigger
         self.days = days
         self.limit = limit
 
-        self.time_start = datetime.datetime.now() - datetime.timedelta(days=days)
-        self.time_end = datetime.datetime.now() - datetime.timedelta(days=0)
-        self.timestamp_start = int(self.time_start.strftime("%s"))
-        self.timestamp_end = int(self.time_end.strftime("%s"))
-        self.mtime_start = mdates.date2num(self.time_start)
-        self.mtime_end = mdates.date2num(self.time_end)
-
-        self.zapi = ZabbixAPI(self.server)
-        self.zapi.login(self.username, self.password)
+        self.timespan = ZabbixTimespan(days)
```

Comment:
Consider passing in the timespan instead of creating the object here (same as you're doing with zapi).